code/modules/reagents/reagents/medical.dm

```diff
@@ -16,6 +16,25 @@
 	scannable = TRUE
 	trait_flags = TACHYCARDIC
 
+/datum/reagent/medicine/inaprovaline/on_mob_add(mob/living/L, metabolism)
+	var/health_ratio = L.health / L.maxHealth
+	var/mob/living/carbon/human/H = L
+	if(TIMER_COOLDOWN_CHECK(L, name))
+		return
+	if(health_ratio < -0.5 && volume > 14) //If you are in crit, and someone injects at least 15u into you at once, you will heal 30% of your physical damage instantly.
+		to_chat(L, span_userdanger("You feel a rush of energy as stimulants course through your veins!"))
+		var/brute_loss = L.getBruteLoss()
+		var/fire_loss = L.getFireLoss()
+		L.adjustBruteLoss(-brute_loss * 0.30)
+		L.adjustFireLoss(-fire_loss * 0.30)
+		L.jitter(5)
+		for(var/datum/internal_organ/I in H.internal_organs)
```

Comment:
```suggestion
		for(var/datum/internal_organ/I AS in H.internal_organs)
```
Ping me so i explain to you the difference

Reply:
done